pkg/controller/atlasstream/atlasstream_instance_controller.go

```diff
@@ -186,3 +192,51 @@ func (r *AtlasStreamsInstanceReconciler) findStreamInstancesForStreamConnection(
 
 	return requests
 }
+
+func (r *AtlasStreamsInstanceReconciler) findStreamInstancesForSecret(ctx context.Context, obj client.Object) []reconcile.Request {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		r.Log.Warnf("watching Secret but got %T", obj)
+		return nil
+	}
+
+	atlasStreamConnections := &akov2.AtlasStreamConnectionList{}
+	listOps := &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(
+			indexer.AtlasStreamConnectionByCredentialsSecret,
+			client.ObjectKeyFromObject(secret).String(),
+		),
+	}
+
+	err := r.Client.List(ctx, atlasStreamConnections, listOps)
+	if err != nil {
+		r.Log.Errorf("failed to list Atlas stream connections: %e", err)
+		return []reconcile.Request{}
+	}
+
+	if len(atlasStreamConnections.Items) == 0 {
```

Comment:
why is this mutually exclusive? if there are no credentials secrets referencing stream instances, there can still be certificate secrets pointing to connections. So if anything, this has to be additive.

Reply:
Clarified externally